connect/runtime/src/test/java/org/apache/kafka/connect/integration/InternalTopicsIntegrationTest.java

```diff
@@ -152,11 +156,20 @@ public void testFailToCreateInternalTopicsWithMoreReplicasThanBrokers() throws I
                     "Body did not contain expected message detailing the worker's in-progress operation: " + body
             );
         }
+
         connect.resetRequestTimeout();
 
+        Future<?> future = worker.getDistributedHerderFuture();
+
+        try {
+            future.get(5000, TimeUnit.MILLISECONDS);
+        }  catch (TimeoutException | ExecutionException exception) {
+            log.error("Failed to start a worker:", exception);
+            future.cancel(true);
+        }
```

Comment:
Few things:
1. We don't need or want to cancel the future; resource cleanup should take place when we shut down the cluster at the end of the test (see [here](https://github.com/apache/kafka/blob/0ada8fac6869cad8ac33a79032cf5d57bfa2a3ea/connect/runtime/src/test/java/org/apache/kafka/connect/integration/InternalTopicsIntegrationTest.java#L61-L65))
2. We should fail the test on a `TimeoutException`, since it indicates that the worker hasn't failed to startup yet
3. We don't need to log anything, especially not at `ERROR` level, if we get an `ExecutionException`
4. We should probably go with a higher timeout; our CI can be really slow sometimes and if we do something too low it might cause flakiness

```suggestion
        assertThrows(
            ExecutionException.class,
            () -> future.get(1, TimeUnit.MINUTES)
        );
```